fs2/model.py

```diff
@@ -256,6 +258,13 @@ def on_load_checkpoint(self, checkpoint):
         Note, this shouldn't fail on different versions of pydantic anymore,
         but it will fail on breaking changes to the config. We should catch those exceptions
         and handle them appropriately."""
+        if "model_info" in checkpoint:
+            assert (
+                checkpoint["model_info"]["name"] == FastSpeech2.__name__
+            ), f"""Wrong model type ({checkpoint["model_info"]["name"]}), we are expecting a { FastSpeech2.__name__ }"""
```

Comment:
Qu'est-ce que l'usager va voir ici? Faudrait que ça soit un mesage d'erreur convivial.

Reply:
Parce qu'ici, un user error va être fréquent...

En fait, je dirais que c'est une `ValueError` ou `TypeError` d'avoir un mismatch, pas un assert failure.

Reply:
I don't think will ever get here unless we have a Frankenstein model where a ckpt has a model of type X but has a config of type Y, then we do `Y.load(ckpt)`  and that would get us to crock on wrong model type.  Otherwise, when we try to load a model `Y` with a chkpt containing a model `X`, this will be caught earlier.
That been said, we should still make it a clearer message just in case.

Reply:
If we don't expect to get here, there's no need to improve the message. As long as we have a good user error message for CLI argument mistakes, the rest is gravy.